service/resource/endpoint/current.go

```diff
@@ -0,0 +1,50 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework/context/canceledcontext"
+
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+
+	pod, err := toPod(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	podAnnotations := pod.GetAnnotations()
+	podIP, ipOk := podAnnotations["endpoint.kvm.giantswarm.io/ip"]
+	podServiceName, serviceNameOk := pod.Annotations["endpoint.kvm.giantswarm.io/service"]
+
+	if !(ipOk && serviceNameOk) {
+		canceledcontext.SetCanceled(ctx)
+		if canceledcontext.IsCanceled(ctx) {
```

Comment:
This does not seem to get hit inside the Test of this function. I can see it when I run this against a cluster though. Is this some magic that can't be unit tested? 

I expected the `SetCancel` to be instant.

Reply:
Check the framework where the functionality for this specific context package is initialized. Check also the context package implementation itself. The signal handling works via channels. I bet in your tests the channel is not set and when you run it the framework does it.

Reply:
Ack, will do. Thanks for the hint!

Reply:
I rabbitholed on this quite a bit, but couldn't do it in a decent manner. The signal handling via channels is so close to the Informer stuff that I couldn't separate it and gave up after it ate too much time.

I might come back to this later but removed the test for now to make more important progress.